### PR TITLE
doc: polish documentation

### PR DESCRIPTION
**Pull request by squakez**
With this PR we have a more up to date documentation.

**Release Note**
```release-note
doc: polish documentation
```

### CODE REVIEW

File: docs/modules/ROOT/pages/configuration/configuration.adoc [hunk 1]
```diff
@@ -1,28 +1,31 @@
...
-[source]
-----
-kamel run --property my.message="Hello World" property-route.yaml
-----
+Camel K does the best it can to discover automatically dependencies required by your application. However, when you're using external code or the operator can't discover dependencies, you'll need to add them manually. See more in the xref:configuration/dependencies.adoc[dependencies] section.
 
-For more details and advanced use cases, see the xref:configuration/runtime-properties.adoc[runtime properties] section.
+Build time properties are the configuration injected during the building of the project. They are mainly used by Camel Quarkus and you may not need at all depending on the kind of applications you're building. See more in the xref:configuration/build-time-properties.adoc[build time properties] section.
+
+If you want to optimize your application to run natively on the cloud, then you may see how to xref:configuration/quarkus-native.adoc[run your application with Quarkus Native].
```
Comment (gansheer) [inline docs/modules/ROOT/pages/configuration/configuration.adoc:19]: What does "Quarkus native" refers to in this instance ? Quarkus with native compilation or the fact that Quarkus is said to be a cloud-native framework ?

**Review by gansheer — APPROVED** [review]
LGTM. Impressive works!

File: docs/modules/ROOT/pages/contributing/local-execution.adoc [hunk 2]
```diff
@@ -62,11 +89,24 @@ IMPORTANT: using build strategy as `Pod` won't probably work as it will expect t
...
+  build:
+    registry:
+      address: localhost:5000
+      insecure: true
+    runtimeVersion: <my-version>
+```
+
+The <my-version> variable must be replaced with the version you are building. For example, `1.3.1-SNAPSHOT`. With these instructions, the operator will pick up and use the snapshot version you have released locally. In order to use the local maven repository, you will also need to edit your IntegrationPlatform as follow:
 ----
 $ k edit ip -n operator-test
```
[the remark this answers is not in the document]
Comment (squakez) [reply docs/modules/ROOT/pages/contributing/local-execution.adoc:111]: Sorry, did not noticed your review. I'll make the change required.

File: docs/modules/ROOT/pages/configuration/configuration.adoc (hunk 1 above)
Comment (squakez) [reply docs/modules/ROOT/pages/configuration/configuration.adoc:19]: Quarkus native compilation

**Pull request merged**